files/common/var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml

```diff
@@ -394,3 +394,10 @@
   with_items:
     - login
     - sshd
+
+#
+# Enable CRA for external variants
+#
+- command: pam-auth-update --enable challenge-response
+  when:
+    - variant is regex("external-.*"):
```

Comment:
It was added in https://github.com/delphix/delphix-platform/pull/110/files

Reply:
> Where does variant come from

From `delphix-entire`; it was recently added: https://github.com/delphix/appliance-build/pull/330

> is that available in the runtime context of the delphix-platform.service service?

Yes. The point of this new field is so that Ansible logic like this can be configured on a per-variant basis (both during the build, and on the running system), without having to reside in appliance-build. The downside of the appliance-build Ansible tasks is they can't be re-run after upgrade.

We've recently made a change to use this variant variable, such that we don't configure the SSH idle timeout for our internal variants; see here: https://github.com/delphix/delphix-platform/pull/110 .. In this specific case, we couldn't use appliance-build to configure the variant, since that appliance-build configuration (applied conditionally based on the variant) would get clobbered on first boot by the delphix-platform configuration (applied unconditionally).

Reply:
That's really nice, thanks for clarifying.